src/main/java/fixie/dictionary_service/service/DictionaryService.java

```diff
@@ -0,0 +1,124 @@
+package fixie.dictionary_service.service;
+
+import fixie.dictionary_service.dto.ActivityDictionaryDTO;
+import fixie.dictionary_service.dto.PartTypeDTO;
+import fixie.dictionary_service.entity.ActivityDictionary;
+import fixie.dictionary_service.entity.PartType;
+import fixie.dictionary_service.repository.ActivityDictionaryRepository;
+import fixie.dictionary_service.repository.PartTypeRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class DictionaryService implements IDictionaryService {
+
+    private final PartTypeRepository partTypeRepository;
+    private final ActivityDictionaryRepository activityDictionaryRepository;
+
+    public DictionaryService(PartTypeRepository partTypeRepository, ActivityDictionaryRepository activityRepository) {
+        this.partTypeRepository = partTypeRepository;
+        this.activityDictionaryRepository = activityRepository;
+    }
+
+    @Override
+    public List<PartType> getPartTypes() {
+
+        return this.partTypeRepository.findAll();
+    }
+
+    @Override
+    public PartType addPartType(PartTypeDTO partTypeDTO) {
+
+
+        PartType partType = PartType.builder()
+                .codeType(partTypeDTO.codeType)
+                .nameType(partTypeDTO.nameType)
+                .build();
+
+        this.partTypeRepository.save(partType);
+
+
+        return partType;
+    }
+
+    @Override
+    public PartType deletePartType(PartTypeDTO partTypeDTO)  {
+
+
+        PartType partType = PartType.builder()
+                .codeType(partTypeDTO.codeType)
+                .nameType(partTypeDTO.nameType)
+                .build();
+
+        this.partTypeRepository.delete(partType);
+
+        return partType;
+    }
+
+    //todo
+    @Override
+    public PartType updatePartType(PartTypeDTO partTypeDTO) {
+
+        PartType partType = PartType.builder()
+                .codeType(partTypeDTO.codeType)
+                .nameType(partTypeDTO.nameType)
+                .build();
```

Comment:
How does it work in your opinion? We want to update something FROM database, not create a new object and then return it.

Reply:
fixed